### PR TITLE
[bitnami/postgresql-ha] Fix confusion between service and container ports

### PR DESCRIPTION
**Pull request by carrodher**
### Description of the change

There is confusion between service ports and container ports that is preventing the chart from working as expected when a non-default or privileged port is set.

#### Current approach
The pgpool pod is being restarted continuously. In the same way, it seems the port 80 is not only being applied to the service but also to the containers:
```console
$ helm install psql bitnami/postgresql-ha --set service.port=80

$ kubectl get pods
NAME                                        READY   STATUS    RESTARTS   AGE
psql-postgresql-ha-pgpool-bbb6b567d-2vxgd   0/1     Running   6          11m
psql-postgresql-ha-postgresql-0             1/1     Running   0          11m
psql-postgresql-ha-postgresql-1             1/1     Running   0          11m

$ kubectl logs -f psql-postgresql-ha-pgpool-bbb6b567d-2vxgd
pgpool 12:39:41.96
pgpool 12:39:41.96 Welcome to the Bitnami pgpool container
pgpool 12:39:41.97 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-pgpool
pgpool 12:39:41.97 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-pgpool/issues
pgpool 12:39:41.97
pgpool 12:39:41.97 INFO  ==> ** Starting Pgpool-II setup **
pgpool 12:39:42.10 INFO  ==> Validating settings in PGPOOL_* env vars...
pgpool 12:39:42.12 INFO  ==> Initializing Pgpool-II...
timeout reached before the port went into state "inuse"
timeout reached before the port went into state "inuse"

# Port 80 in the services
#
$ kubectl get svc
NAME                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
psql-postgresql-ha-pgpool                ClusterIP   10.30.243.108   <none>        80/TCP    13m
psql-postgresql-ha-postgresql            ClusterIP   10.30.243.252   <none>        80/TCP    13m
psql-postgresql-ha-postgresql-headless   ClusterIP   None            <none>        80/TCP    13m

# Port 80 is applied to the 3 containers
#
$ kubectl describe pod psql-postgresql-ha-pgpool-bbb6b567d-2vxgd | grep Port
    Port:           80/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-postgresql-0 | grep Port
    Port:           80/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-postgresql-1 | grep Port
    Port:           80/TCP
    Host Port:      0/TCP
```

#### With this PR
```console
$ helm install psql . --set service.port=80

$ kubectl get pods
NAME                                         READY   STATUS    RESTARTS   AGE
psql-postgresql-ha-pgpool-5497cdcdcb-nshsj   1/1     Running   0          2m1s
psql-postgresql-ha-postgresql-0              1/1     Running   0          2m1s
psql-postgresql-ha-postgresql-1              1/1     Running   0          2m1s

$ kubectl logs -f psql-postgresql-ha-pgpool-5497cdcdcb-nshsj
pgpool 12:41:59.39
pgpool 12:41:59.39 Welcome to the Bitnami pgpool container
pgpool 12:41:59.41 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-pgpool
pgpool 12:41:59.42 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-pgpool/issues
pgpool 12:41:59.42
pgpool 12:41:59.42 INFO  ==> ** Starting Pgpool-II setup **
pgpool 12:41:59.47 INFO  ==> Validating settings in PGPOOL_* env vars...
pgpool 12:41:59.50 INFO  ==> Initializing Pgpool-II...
pgpool 12:42:29.76 INFO  ==> Generating pg_hba.conf file...
pgpool 12:42:29.77 INFO  ==> Generating pgpool.conf file...
pgpool 12:42:30.03 DEBUG ==> Adding 'psql-postgresql-ha-postgresql-0.psql-postgresql-ha-postgresql-headless' information to the configuration...
pgpool 12:42:30.04 DEBUG ==> Adding 'psql-postgresql-ha-postgresql-1.psql-postgresql-ha-postgresql-headless' information to the configuration...
pgpool 12:42:30.05 INFO  ==> Generating password file for local authentication...
pgpool 12:42:30.06 INFO  ==> Generating password file for pgpool admin user...
pgpool 12:42:30.08 INFO  ==> ** Pgpool-II setup finished! **
pgpool 12:42:30.15 INFO  ==> ** Starting Pgpool-II **
...

# Port 80 in the services except headless
#
$ kubectl get svc
NAME                                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
psql-postgresql-ha-pgpool                ClusterIP   10.30.254.9    <none>        80/TCP     3m34s
psql-postgresql-ha-postgresql            ClusterIP   10.30.252.23   <none>        80/TCP     3m34s
psql-postgresql-ha-postgresql-headless   ClusterIP   None           <none>        5432/TCP   3m34s

# Port 5432 is applied to the 3 containers
#
$ kubectl describe pod psql-postgresql-ha-postgresql-0 | grep Port
    Port:           5432/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-postgresql-1 | grep Port
    Port:           5432/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-pgpool-5497cdcdcb-nshsj | grep Port
    Port:           5432/TCP
    Host Port:      0/TCP
```

### CODE REVIEW

**Comment by carrodher** [comment]
```console
$ helm install psql bitnami/postgresql-ha --set service.port=80

$ kubectl get pods
NAME                                        READY   STATUS    RESTARTS   AGE
psql-postgresql-ha-pgpool-bbb6b567d-2vxgd   0/1     Running   6          11m
psql-postgresql-ha-postgresql-0             1/1     Running   0          11m
psql-postgresql-ha-postgresql-1             1/1     Running   0          11m

# Port 80 in the services
#
$ kubectl get svc
NAME                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
psql-postgresql-ha-pgpool                ClusterIP   10.30.243.108   <none>        80/TCP    13m
psql-postgresql-ha-postgresql            ClusterIP   10.30.243.252   <none>        80/TCP    13m
psql-postgresql-ha-postgresql-headless   ClusterIP   None            <none>        80/TCP    13m

# Port 80 is applied to the 3 containers
#
$ kubectl describe pod psql-postgresql-ha-pgpool-bbb6b567d-2vxgd | grep Port
    Port:           80/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-postgresql-0 | grep Port
    Port:           80/TCP
    Host Port:      0/TCP

$ kubectl describe pod psql-postgresql-ha-postgresql-1 | grep Port
    Port:           80/TCP
    Host Port:      0/TCP
```

File: bitnami/postgresql-ha/values.yaml [hunk 1]
```diff
@@ -154,6 +154,9 @@ postgresql:
   ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies
   ##
   updateStrategyType: RollingUpdate
+  ## @param postgresql.containerPort PostgreSQL port
+  ##
+  containerPort: 5432
```
Comment (juan131) [inline bitnami/postgresql-ha/values.yaml:159]: I'm missing the equivalent (`pgpool.containerPort`) for Pgpool, right?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks Carlos!! Great PR, please check my comments.

File: bitnami/postgresql-ha/values.yaml (hunk 1 above)
Comment (carrodher) [reply bitnami/postgresql-ha/values.yaml:159]: I just added it to the values, README and pgpool deployment

**Review by juan131 — APPROVED** [review]
LGTM

**Review by beltran-rubo — APPROVED** [review]

**Pull request merged**